src/middlewared/middlewared/plugins/account.py

```diff
@@ -846,7 +846,8 @@ def recreate_homedir_if_not_exists(self, user, group, mode):
                 }).wait_sync(raise_error=True)
 
     @api_method(UserDeleteArgs, UserDeleteResult, audit='Delete user', audit_callback=True)
-    def do_delete(self, audit_callback, pk, options):
+    @pass_app(require=True)
```

Comment:
We probably need to allow REST here. `require` means that we can't have internal callers to `user.delete` which is also probably not intended.

Reply:
`require` just means that `app` cannot be `None`, no? We can pass `app` in our internal calls like I do on line 877.

Reply:
> require just means that app cannot be None, no?

Right. And there may be valid internal callers for it where we don't want to pass the authenticated credentials.

Reply:
If you don't set `rest=True` then method is unavailable over REST this breaks REST API access to user deletion.